File: src/01-easy/temperatures/solution.ts
const n = parseInt(readline());
if (n === 0) { console.log(0); } else {
    const t = readline().split(' ').map(Number);
    let r = t[0];
    for (const v of t) { if (Math.abs(v) < Math.abs(r) || (Math.abs(v) === Math.abs(r) && v > 0)) r = v; }
    console.log(r);
}

File: src/01-easy/temperatures/solution.clj
(ns Player
  (:gen-class))
(defn -main [& args]
  (let [n (Integer/parseInt (read-line))]
    (if (zero? n) (println 0)
      (let [t (map #(Integer/parseInt %) (.split (.trim (read-line)) " "))
            closer (fn [a b] (if (or (< (Math/abs b) (Math/abs a))
                                     (and (= (Math/abs b) (Math/abs a)) (pos? b))) b a))]
        (println (reduce closer t))))))
